app/main/views/broadcast.py

```diff
@@ -135,22 +135,46 @@ def new_broadcast(service_id):
 def write_new_broadcast(service_id):
     form = BroadcastTemplateForm()
 
+    broadcast_message_id = request.args.get("broadcast_message_id")
+    broadcast_message = None
+
     if form.validate_on_submit():
-        broadcast_message = BroadcastMessage.create_from_content(
-            service_id=current_service.id,
-            content=form.template_content.data,
-            reference=form.name.data,
-        )
+        if broadcast_message_id:
+            BroadcastMessage.update_from_content(
+                service_id=current_service.id,
+                broadcast_message_id=broadcast_message_id,
+                content=form.template_content.data,
+                reference=form.name.data,
+            )
+        else:
```

Comment:
Could this be used with parameter changing to update a broadcast message which has already been sent, should we only allow them to update draft messages?